include/maths/CTreeShapFeatureImportance.h

```diff
@@ -50,67 +50,89 @@ class MATHS_EXPORT CTreeShapFeatureImportance {
 private:
     using TSizeVec = std::vector<std::size_t>;
 
-    //! Manages variables for the current path through the tree as the main algorithm proceeds.
-    struct SPath {
-        explicit SPath(std::size_t length)
-            : s_FractionOnes(length), s_FractionZeros(length),
-              s_FeatureIndex(length, -1), s_Scale(length), s_MaxLength(length) {}
-
-        void extend(int featureIndex, double fractionZero, double fractionOne) {
-            if (s_NextIndex < s_MaxLength) {
-                s_FeatureIndex[s_NextIndex] = featureIndex;
-                s_FractionZeros[s_NextIndex] = fractionZero;
-                s_FractionOnes[s_NextIndex] = fractionOne;
-                if (s_NextIndex == 0) {
-                    s_Scale[s_NextIndex] = 1.0;
-                } else {
-                    s_Scale[s_NextIndex] = 0.0;
-                }
-                ++s_NextIndex;
-            }
+    //! Collects the elements of the path through decision tree that are updated together
+    struct SPathElement {
+        double s_FractionOnes;
+        double s_FractionZeros;
+        int s_FeatureIndex;
```

Comment:
I'd default initialise these: even though they're overwritten it is probably better to have them properly initialised (shouldn't cost anything since this only happens once), e.g.
```suggestion
        int s_FeatureIndex = -1;
```
etc

Reply:
Fixed